src/main/java/com/google/vinet/data/MatchingRunner.java

```diff
@@ -0,0 +1,194 @@
+    return fetchVolunteerTimeSlots(this.date, this.datastore);
+  }
+
+  /**
+   * Get a PreparedQuery which will return all TimeSlots with the date and entity type provided when
+   * executed using the provided DataStore implementation.
+   *
+   * @param entityName The entity name of the TimeSlots to be fetched.
+   * @param date The date that the fetched TimeSlots must be scheduled on.
+   * @param datastore The DataStore implementation to be used when preparing the query.
+   * @return A PreparedQuery which will return all TimeSlots with the date and entity type provided
+   * when executed using the provided DataStore implementation.
+   */
+  protected static PreparedQuery getTimeSlotsQuery(
+      String entityName, LocalDate date, DatastoreService datastore) {
+    final Query query = new Query(entityName);
+
+    final Filter dateFilter = new FilterPredicate("date", FilterOperator.EQUAL, date.toString());
+
+    query.setFilter(dateFilter);
+
+    return datastore.prepare(query);
+  }
+
+  /**
+   * Fetch all IsolateTimeSlots scheduled for the provided date using the provided DataStore implementation.
+   * @param date The date to filter the IsolateTimeSlots by.
+   * @param datastore The DataStore implementation to be queried.
+   * @return All IsolateTimeSlots scheduled for the provided date using the provided DataStore implementation.
+   */
+  protected static Set<IsolateTimeSlot> fetchIsolateTimeSlots(LocalDate date, DatastoreService datastore) {
+    PreparedQuery preparedQuery =  getTimeSlotsQuery("IsolateTimeSlot", date, datastore);
+
+    final FetchOptions fetchOptions = FetchOptions.Builder.withDefaults();
+
+    final Set<IsolateTimeSlot> timeSlots = new HashSet<>();
+
+    for (Entity entity : preparedQuery.asIterable(fetchOptions)) {
+      final Isolate isolate = new Isolate((String) entity.getProperty("isolateId"));
+      final Key ticketKey = KeyFactory.stringToKey((String) entity.getProperty("ticketKey"));
+      final LocalDate localDate = LocalDate.parse((String) entity.getProperty("date"));
+      final Instant start = Instant.parse((String) entity.getProperty("startTime"));
+      final Instant end = Instant.parse((String) entity.getProperty("endTime"));
+      final IsolateTimeSlot isolateTimeSlot = new IsolateTimeSlot(start, end, isolate, localDate, ticketKey);
+      timeSlots.add(isolateTimeSlot);
+    }
+
+    return timeSlots;
+  }
+
+  /**
+   * Fetch all VolunteerTimeSlots scheduled for the provided date using the provided DataStore implementation.
+   * @param date The date to filter the VolunteerTimeSlots by.
+   * @param datastore The DataStore implementation to be queried.
+   * @return All VolunteerTimeSlots scheduled for the provided date using the provided DataStore implementation.
+   */
+  protected static Set<VolunteerTimeSlot> fetchVolunteerTimeSlots(LocalDate date, DatastoreService datastore) {
+    final PreparedQuery preparedQuery = getTimeSlotsQuery("volunteer_timeslots", date, datastore);
```

Comment:
nit: IsolateTimeSlot and volunteer_timeslot is inconsistent, can you use the same case style?

Reply:
Since I wasn't the owner of the volunteer_timeslots feature, and the naming convention wasn't introduced in this PR, I'm not going to implement renaming here, but I have added the issue so we can track it (See issue #12)